src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCallerEngine.java

```diff
@@ -711,6 +711,8 @@ public void shutdown() {
             }
         }
 
+        // Write assembly region debug output if present
+        assemblyEngine.printDebugHistograms();
```

Comment:
`shutdown()` is usually for closing resources -- is this the best place for this call?

Also, how much memory will this feature use if activated for large-scale runs?

Reply:
The memory should be negligable. The histograms both have bin sizes of 1, so its essentially a map of integer values to a count. The first map is bounded by the K we use for k best haplotype finder (so 256), and the second is bounded by the number of kmer sizes we use (so 6 i think currently?) I'm not at all worried. 

As for the on-shutdown I needed somewhere to print this thing because you unfortunately can't really print a histogram on the fly so it needs to be at the end of time somewhere. I'm not really sure where else is the right place for it, because the engine is what owns the assemblyEngine where this debug info is being stored.